ziti/src/main/kotlin/org/openziti/impl/ZitiContextImpl.kt

```diff
@@ -233,15 +233,36 @@ internal class ZitiContextImpl(internal val id: Identity, enabled: Boolean) : Zi
     }
 
     private suspend fun runner() {
+        val ctrlEndpoints = id.controllers().toMutableSet()
+        lateinit var authenticator: ZitiAuthenticator
         while(true) {
-            val session = runCatching { login() }.getOrElse {
-                e("failed to login, cannot continue")
-                updateStatus(ZitiContext.Status.NotAuthorized(it))
-                throw it
+            val c = Retry.withExponentialBackoff {
```

Comment:
just pointing out that it seems a bit strange to have a while(true) when the withExponentialBackoff func runs forever? Maybe it makes sense to do all of this in the same lambda?

Reply:
exponential backoff is only for finding a controller that is online